crates/ide_assists/src/handlers/replace_if_let_with_match.rs

```diff
@@ -53,17 +54,30 @@ pub(crate) fn replace_if_let_with_match(acc: &mut Assists, ctx: &AssistContext)
     });
     let scrutinee_to_be_expr = if_expr.condition()?.expr()?;
 
-    let mut pat_bodies = Vec::new();
+    let mut pat_seen = false;
+    let mut cond_bodies = Vec::new();
     for if_expr in if_exprs {
         let cond = if_expr.condition()?;
         let expr = cond.expr()?;
-        if scrutinee_to_be_expr.syntax().text() != expr.syntax().text() {
-            // Only if all condition expressions are equal we can merge them into a match
-            return None;
-        }
-        let pat = cond.pat()?;
+        let cond = match cond.pat() {
+            Some(pat) => {
+                if scrutinee_to_be_expr.syntax().text() != expr.syntax().text() {
+                    // Only if all condition expressions are equal we can merge them into a match
+                    return None;
+                } else {
```

Comment:
No need for an else here.

Could you create an issue to write a weak warning diagnostic and fixit for unnecessary else, while we are at it? :)

Reply:
#9457